src/main/java/apoc/export/cypher/MultiStatementCypherSubGraphExporter.java

```diff
@@ -19,42 +21,90 @@
  * Idea is to lookup nodes for relationships via a unqiue index
  * either one inherent to the original node, or a artificial one that indexes the original node-id
  * and which is removed after the import.
- *
+ * <p>
  * Outputs indexes and constraints at the beginning as their own transactions
  */
 public class MultiStatementCypherSubGraphExporter {
+    private final static String UNIQUE_ID_LABEL = "UNIQUE IMPORT LABEL";
+    private final static String Q_UNIQUE_ID_LABEL = quote(UNIQUE_ID_LABEL);
+    private final static String UNIQUE_ID_PROP = "UNIQUE IMPORT ID";
     private final SubGraph graph;
     private final Map<String, String> uniqueConstraints;
     Set<String> indexNames = new LinkedHashSet<>();
     Set<String> indexedProperties = new LinkedHashSet<>();
-    private final static String UNIQUE_ID_LABEL = "UNIQUE IMPORT LABEL";
-    private final static String Q_UNIQUE_ID_LABEL = quote(UNIQUE_ID_LABEL);
-    private final static String UNIQUE_ID_PROP = "UNIQUE IMPORT ID";
     private long artificialUniques = 0;
 
     public MultiStatementCypherSubGraphExporter(SubGraph graph) {
         this.graph = graph;
         uniqueConstraints = gatherUniqueConstraints(indexNames, indexedProperties);
     }
 
-    public void export(PrintWriter out, int batchSize, Reporter reporter) {
+    public static String quote(String id) {
+        return "`" + id + "`";
+    }
+
+    public static String label(String id) {
+        return ":`" + id + "`";
+    }
+
+    /**
+     * @param fileName
+     * @param suffix
+     * @return
+     */
+    private String normalizeFileName(final String fileName, String suffix) {
+        // TODO check if this should be follow the same rules of FileUtils.readerFor
```

Comment:
This should be smarter than now.
Accept an URI and follow the same logic of FileUtils.readerFor